src/ServiceManagement/Common/Commands.ScenarioTest/Resources/ServiceManagement/ServiceManagementTests.ps1

```diff
@@ -1007,3 +1007,61 @@ function Run-RedeployVirtualMachineTest
         Cleanup-CloudService $svcName;
     }
 }
+
+# Test Initiate Maintenance VM
+function Run-InitiateMaintenanceTest
+{
+	# Depending on the environment, the initiate maintenance operation may return 200
+	# or 400 with error message like "User initiated maintenance on the virtual machine was
+	# successfully completed". Both are expected reponse.
+    # To continue script, $ErrorActionPreference should be set to 'SilentlyContinue'.
+	$tempErrorActionPreference = $ErrorActionPreference;
+    $ErrorActionPreference='SilentlyContinue';
+	
+	# Setup
+	$location = "Central US EUAP";
+	$imgName = Get-DefaultImage $location;
+
+	$storageName = 'pstest' + (getAssetName);
+	New-AzureStorageAccount -StorageAccountName $storageName -Location $location;
+
+	# Associate the new storage account with the current subscription
+	Set-CurrentStorageAccountName $storageName;
+
+	$vmName = "psvm01";
+	$svcName = 'pstest' + (Get-CloudServiceName);
+	$userName = "pstestuser";
+	$password = $PLACEHOLDER;
+
+	# Test
+	New-AzureService -ServiceName $svcName -Location $location;
+
+	try
+	{
+		New-AzureQuickVM -Windows -ImageName $imgName -Name $vmName -ServiceName $svcName -AdminUsername $userName -Password $password;
+		Start-Sleep -s 300;
```

Comment:
Commented out this line for Playback mode such as below:

#Start-Sleep -s 300; #Uncomment this line for record mode testing.